src/Feature.php

```diff
@@ -8,89 +8,109 @@
 
 namespace FriendsOfCat\LaravelFeatureFlags;
 
-use Feature\Contracts\World;
+use Illuminate\Support\Facades\Auth;
 
-class Feature implements World
+class Feature
 {
+    const ON  = "on";
+    const OFF = "off";
 
     /**
-     * @param string $name
-     * @param mixed $default
-     * @return mixed
+     * @var self
      */
-    public function configValue($name, $default = null)
-    {
-        // TODO: Implement configValue() method.
-    }
+    private static $instance;
 
     /**
-     * @return mixed
+     * @var array
      */
-    public function uaid()
-    {
-        // TODO: Implement uaid() method.
-    }
+    private $stanza;
 
     /**
-     * @return int
+     * @param array $stanza
      */
-    public function userId()
+    private function __construct(array $stanza)
     {
-        // TODO: Implement userId() method.
+        $this->stanza = $stanza;
     }
 
     /**
-     * @param string|int $userId
-     * @return string
+     * @param array $stanza
+     * @return Feature
      */
-    public function userName($userId)
+    public static function create(array $stanza)
     {
-        // TODO: Implement userName() method.
+        if (static::$instance) {
+            return static::$instance;
+        }
+
+        static::$instance = new static($stanza);
+
+        return static::$instance;
     }
 
+
     /**
-     * @param int $userId
-     * @param int $groupdId
+     * @param $feature
      * @return bool
      */
-    public function inGroup($userId, $groupdId)
+    public static function isEnabled($feature)
     {
-        // TODO: Implement inGroup() method.
+        $feature_variant = static::getConfig($feature);
+
+        if ($feature_variant != self::ON and $feature_variant != self::OFF) {
+            return self::isUserEnabled($feature_variant);
+        }
+
+        return ($feature_variant == self::ON);
     }
 
     /**
-     * @param int $userId
-     * @return bool
+     * @param $feature
+     * @return string
      */
-    public function isAdmin($userId)
+    private static function getConfig($feature)
```

Comment:
@luiz-albertoni so we want to have 0 static calls by the time we are done. This should be just normal PHP taking advantage of dependency injection, cache, state etc.